src/coreclr/vm/debuginfostore.cpp

```diff
@@ -680,26 +701,77 @@ PatchpointInfo * CompressDebugInfo::RestorePatchpointInfo(IN PTR_BYTE pDebugInfo
     }
     CONTRACTL_END;
 
-    PTR_PatchpointInfo patchpointInfo = NULL;
-
     // Check flag byte.
     BYTE flagByte = *pDebugInfo;
     pDebugInfo++;
 
-    if (flagByte == 1)
+    if ((flagByte & EXTRA_DEBUG_INFO_PATCHPOINT) == 0)
+        return NULL;
+
+    return static_cast<PatchpointInfo*>(PTR_READ(dac_cast<TADDR>(pDebugInfo), dac_cast<PTR_PatchpointInfo>(pDebugInfo)->PatchpointInfoSize()));
```

Comment:
This struct has a flexible array at the end, IIUC the previous code was not correct in DAC builds.